src/Middleware/StaticFiles/src/StaticFileContext.cs

```diff
@@ -47,6 +43,11 @@ internal struct StaticFileContext
 
         public StaticFileContext(HttpContext context, StaticFileOptions options, ILogger logger, IFileProvider fileProvider, string? contentType, PathString subPath)
         {
+            if (!subPath.HasValue)
```

Comment:
@Tratcher `StaticFileMiddleware.ValidatePath(...)` returns `true` when it sets `subPath` to `PathString.Empty`. In turn, `PathString.Empty` shows `HasValue == false` because that property uses `!string.IsNullOrEmpty(Value)`. This feels like a bug -- an empty value is still a value and should get passed this check. Thoughts❔

For now, I'm going to change `PathString.HasValue` but I suspect my next iteration won't fix all the test failures. You have a bit of time ☹️

Reply:
Changed my mind because the result is extra allocations e.g. `UsePathBaseExtensions.UsePathBase(...)` will add an unnecessary `UsePathBaseMiddleware`. Will instead change this to check `subPath.Value == null`.